### PR TITLE
Fixed issue on rename

### PR DESCRIPTION
**Pull request by patduin**

### CODE REVIEW

File: circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperationTest.java [hunk 1]
```diff
@@ -55,8 +55,8 @@ public void setUp() throws TException {
     toTable.setTableName(TO_TABLE_NAME);
     fromTable.setDbName(FROM_DB_NAME);
     toTable.setDbName(TO_DB_NAME);
-    when(client.getTable(FROM_DB_NAME, FROM_TABLE_NAME)).thenReturn(fromTable);
-    when(client.getTable(TO_DB_NAME, TO_TABLE_NAME)).thenReturn(toTable);
+    when(client.getTable(FROM_DB_NAME, FROM_TABLE_NAME)).thenReturn(new Table(fromTable));
```
Comment (patduin) [inline circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperationTest.java:58]: these test were wrong cause the "test" table object was injected into the real code and then mutated so the test weren't testing what they thought they were testing. Was never failing.

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java [hunk 2]
```diff
@@ -40,20 +40,27 @@ public RenameTableOperation(DropTableService dropTableService) {
...
+    String fromDatabaseName = fromTable.getDbName();
     String fromTableName = fromTable.getTableName();
     String toTableName = toTable.getTableName();
     String toDelete = toTableName + DELETE_ME;
     try {
+      fromTable.setDbName(toTable.getDbName());
       fromTable.setTableName(toTableName);
       toTable.setTableName(toDelete);
+      LOG
+          .info("Alter table {}.{} to {}.{}", toTable.getDbName(), toTableName, toTable.getDbName(),
```
Comment (massdosage) [inline circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:55]:
```suggestion
          .info("Renaming table {}.{} to {}.{}", toTable.getDbName(), toTableName, toTable.getDbName(),
```

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java [hunk 3]
```diff
@@ -40,20 +40,27 @@ public RenameTableOperation(DropTableService dropTableService) {
...
+              toTable.getTableName());
       client.alter_table(toTable.getDbName(), toTableName, toTable);
-      client.alter_table(fromTable.getDbName(), fromTableName, fromTable);
+      LOG
+          .info("Alter table {}.{} to {}.{}", fromDatabaseName, fromTableName, fromTable.getDbName(),
```
Comment (massdosage) [inline circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:59]:
```suggestion
          .info("Renamed table {}.{} to {}.{}", fromDatabaseName, fromTableName, fromTable.getDbName(),
```

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java (hunk 2 above)
Comment (massdosage) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:55]: Also, shouldn't the first two arguments be the "from" values? Same in the log statement below.

Comment (massdosage) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:55]: Alternatively we remove these two new log statements and just keep the one we already had.

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java [hunk 4]
```diff
@@ -40,20 +40,27 @@ public RenameTableOperation(DropTableService dropTableService) {
...
+              fromTable.getTableName());
+      client.alter_table(fromDatabaseName, fromTableName, fromTable);
     } finally {
+      LOG.info("Drop table {}.{}", toTable.getDbName(), toDelete);
```
Comment (massdosage) [inline circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:63]:
```suggestion
      LOG.info("Dropping table {}.{}", toTable.getDbName(), toDelete);
```

**Review by massdosage — CHANGES_REQUESTED** [review]

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java [hunk 5]
```diff
@@ -40,20 +40,27 @@ public RenameTableOperation(DropTableService dropTableService) {
...
     Table toTable = client.getTable(to.getDbName(), to.getTableName());
+    String fromDatabaseName = fromTable.getDbName();
     String fromTableName = fromTable.getTableName();
     String toTableName = toTable.getTableName();
     String toDelete = toTableName + DELETE_ME;
     try {
+      fromTable.setDbName(toTable.getDbName());
       fromTable.setTableName(toTableName);
       toTable.setTableName(toDelete);
+      LOG
```
Comment (JayGreeeen) [inline circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:54]: Might be worth adding in the comment lines above the logs like we have in Cloverleaf? https://github.expedia.biz/eg-data-lake/cloverleaf/blob/5c948f54c72558260a6744a9d4a752882d61ea99/cloverleaf-steps/src/main/java/com/hotels/bdp/cloverleaf/hive/table/RenameTableOperation.java I find it a bit easier to follow

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java [hunk 6]
```diff
@@ -40,20 +40,27 @@ public RenameTableOperation(DropTableService dropTableService) {
...
   public void execute(CloseableMetaStoreClient client, Table from, Table to) throws Exception {
-    LOG
-        .info("Renaming table {}.{} to {}.{}", from.getDbName(), from.getTableName(), to.getDbName(),
-            to.getTableName());
+    LOG.info("Renaming table {}.{} to {}.{}", from.getDbName(), from.getTableName(), to.getDbName(), to.getTableName());
     Table fromTable = client.getTable(from.getDbName(), from.getTableName());
     Table toTable = client.getTable(to.getDbName(), to.getTableName());
+    String fromDatabaseName = fromTable.getDbName();
     String fromTableName = fromTable.getTableName();
     String toTableName = toTable.getTableName();
```
Comment (JayGreeeen) [inline circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:48]: Worth also having `String toDatabaseName = toTable.getDbName()` ?  
toTable.getDbName() is called quite a few times

**Review by nvitucci — APPROVED** [review]

**Comment by nvitucci** [comment]
Looks good, provided log messages are double-checked as requested by the other reviewers.

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java (hunk 6 above)
Comment (patduin) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:48]: sure

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java (hunk 5 above)
Comment (patduin) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:54]: ok

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java (hunk 4 above)
Comment (patduin) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:63]: done

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java (hunk 3 above)
Comment (patduin) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:59]: that's wrong, I kept the "alter" but changed to "Altering"
If you run the tests the logs read like this and are very clear:
```
- Renaming table old_db.old_table to new_db.new_table
- Altering table new_db.new_table to new_db.new_table_delete_me
- Altering table old_db.old_table to new_db.new_table
- Dropping table new_db.new_table_delete_me
```

File: circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java (hunk 2 above)
Comment (patduin) [reply circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/RenameTableOperation.java:55]: It's more clear when you actually read the log messages in test and when running CT.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
+## [TBD] - TBD
```
Comment (JayGreeeen) [inline CHANGELOG.md:1]: version number?

Comment (massdosage) [reply CHANGELOG.md:1]: Yup, 16.3.3 I guess?

Comment (patduin) [reply CHANGELOG.md:1]: can go in next release, this is a "safety" fix we're not using different database in Circus Train though.

Comment (massdosage) [reply CHANGELOG.md:1]: OK, fine, just leave it then.

**Review by massdosage — APPROVED** [review]

**Review by JayGreeeen — APPROVED** [review]

**Pull request merged**